### ISSUE TITLE
More Cosmos DB model customization options

### ISSUE THREAD

**Issue by vmelamed** [dotnet/efcore#32517]
Are there any plans to add these provider-specific modeling options:
- auto/manual scaling and max RU/s for a DB or container 
- specify max RU/s for a DB or container
- indexing options
  - included
  - excluded
  - unique

**Comment by roji** [comment]
Note #34023 - we were previously simply ignoring HasIndex(), since 9.0 we throw.

This issue would be about exposing ways for the user to define the container's [indexing policy](https://learn.microsoft.com/en-us/azure/cosmos-db/index-policy). By default we'd stay as today, with the Cosmos automatic indexing of all properties (and HasIndex() would continue throwing). But if the user opts into defining the policies, HasIndex() could at that point define the indexing policy.

Note that this would only affect EnsureCreated(). This might not be very valuable, as users can configure their container indexing policy outside of EF anyway, and it's not clear how many people actually create their containers via EF.

**Comment by WhitWaldo** [comment]
I would certainly appreciate seeing this feature added to EF so that I know when containers are created by Cosmos, they're created uniformly regardless of the current target.  Yes, it'd generally be something that only happened a few times at best, but having that built-in consistency would be helpful.

**Comment by roji** [comment]
Note my comment above - this is likely not very valuable since containers can (and probably usually are) created outside of EF anyway. EF probably won't be able to cover all of the container configuration/setup options anyway.

**Comment by WhitWaldo** [comment]
@roji Certainly, but that's going to be true of any ORM, especially for a mostly one-time operation like this. My point in chiming in is just that it'd be nice to ensure that on that rare occasion that my application creates containers somewhere, it'd be nice to have that guarantee that the desired indexes are there as opposed to always adding a manual step after the otherwise automatic creation.

Personally, I'd get the most value out of just having Exclude support since Cosmos is automatically all-inclusive.
